packages/twenty-front/src/modules/object-record/record-table/components/RecordTableRecordGroupRows.tsx

```diff
@@ -1,37 +1,64 @@
 import { useCurrentRecordGroupId } from '@/object-record/record-group/hooks/useCurrentRecordGroupId';
-import { recordIndexAllRowIdsComponentState } from '@/object-record/record-index/states/recordIndexAllRowIdsComponentState';
-import { recordIndexRowIdsByGroupComponentFamilyState } from '@/object-record/record-index/states/recordIndexRowIdsByGroupComponentFamilyState';
+import { recordIndexRecordIdsByGroupComponentFamilyState } from '@/object-record/record-index/states/recordIndexRecordIdsByGroupComponentFamilyState';
+import { recordIndexAllRecordIdsComponentSelector } from '@/object-record/record-index/states/selectors/recordIndexAllRecordIdsComponentSelector';
 import { RecordTableRow } from '@/object-record/record-table/record-table-row/components/RecordTableRow';
+import { isRecordGroupTableSectionToggledComponentState } from '@/object-record/record-table/record-table-section/states/isRecordGroupTableSectionToggledComponentState';
 import { useRecoilComponentFamilyValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentFamilyValueV2';
 import { useRecoilComponentValueV2 } from '@/ui/utilities/state/component-state/hooks/useRecoilComponentValueV2';
+import { AnimatePresence, motion } from 'framer-motion';
 import { useMemo } from 'react';
+import { isDefined } from '~/utils/isDefined';
+
+const MotionRecordTableRow = motion(RecordTableRow);
 
 export const RecordTableRecordGroupRows = () => {
-  const recordGroupId = useCurrentRecordGroupId();
+  const currentRecordGroupId = useCurrentRecordGroupId();
+
+  const allRecordIds = useRecoilComponentValueV2(
+    recordIndexAllRecordIdsComponentSelector,
+  );
 
-  const allRowIds = useRecoilComponentValueV2(
-    recordIndexAllRowIdsComponentState,
+  const recordIdsByGroup = useRecoilComponentFamilyValueV2(
+    recordIndexRecordIdsByGroupComponentFamilyState,
+    currentRecordGroupId,
   );
 
-  const recordGroupRowIds = useRecoilComponentFamilyValueV2(
-    recordIndexRowIdsByGroupComponentFamilyState,
-    recordGroupId,
+  const isRecordGroupTableSectionToggled = useRecoilComponentFamilyValueV2(
+    isRecordGroupTableSectionToggledComponentState,
+    currentRecordGroupId,
   );
 
   const rowIndexMap = useMemo(
-    () => new Map(allRowIds.map((id, index) => [id, index])),
-    [allRowIds],
+    () => new Map(allRecordIds.map((recordId, index) => [recordId, index])),
+    [allRecordIds],
   );
 
-  return recordGroupRowIds.map((recordId) => {
-    const rowIndex = rowIndexMap.get(recordId);
+  // TODO: Animation is not working, find a way to make it works
+  const variants = {
+    hidden: { height: 0, opacity: 0 },
```

Comment:
if it's not working, should we commit that?

Reply:
It's not working but actually not doing anything strange visually, I've pushed it, so it's easier to continue the ticket about the animation

Reply:
ok, i'm fine with TODOs